### PR TITLE
bots: Use external test learning code

### PR DESCRIPTION
**Pull request by stefwalter**
This uses containers deployed from cockpituous/learn in the various
clusters to perform teh machine learning tasks, rather than having
all that code living here in the bots/ directory.

 * [x] learn-tests http://cockpit-learn.cockpit.svc.cluster.local/train/tests-train-1.jsonl.gz
 * [x] learn-tests http://cockpit-learn.verify.svc.cluster.local/train/tests-train-1.jsonl.gz
 * [x] #10278

### CODE REVIEW

**Comment by cockpituous** [comment]
tests-data in progress on centosci-tasks-h5v1g.
Log: http://fedorapeople.org/groups/cockpit/logs/tests-data-10276-20181011-114444/

**Comment by cockpituous** [comment]
Task threw an exception: http://fedorapeople.org/groups/cockpit/logs/tests-data-10276-20181011-114444/

**Comment by martinpitt** [comment]
@stefwalter: How should this be interpreted: https://cockpit-learn.cockpit.svc.cluster.localNone/train/tests-train-1.jsonl.gz

Aside from the "None", which is most certainly wrong, this doesn't really describe a particular cluster such as e2e.bos.redhat.com or console.apps.ci.centos.org. This URL is invalid outside of a cluster, and valid in any cluster. Is that by intent?

**Comment by stefwalter** [comment]
> @stefwalter: How should this be interpreted: https://cockpit-learn.cockpit.svc.cluster.localNone/train/tests-train-1.jsonl.gz

There's a bug there, obviously with the None. I'll fix that.

The goal is that this includes the project/namespace DNS name "cockpit.svc.cluster.local" and thus can only run from within the cluster that has that project.

I can see of no way to identify the cluster itself from within itself. So I chose to do it by project/namespace name given that we have different project names in different clusters. But if you have a better idea, I'd be happy to change this.

**Comment by cockpituous** [comment]
tests-data in progress on bos-tasks-sl1cz.
Log: http://fedorapeople.org/groups/cockpit/logs/tests-data-10276-20181011-123232/

**Comment by martinpitt** [comment]
semaphore fails:

    FAIL: tools/test-static-code 1 pyflakes bots pkg tools

**Comment by stefwalter** [comment]
Fixed static code checks.

**Comment by martinpitt** [comment]
> So I chose to do it by project/namespace name given that we have different project names in different clusters. But if you have a better idea, I'd be happy to change this.

For the record: I don't. I was just misled by the commit message part where it talks about "a specific cluster". I understand it now, and I don't see a better solution either. Thanks!

**Comment by cockpituous** [comment]
Task threw an exception: http://fedorapeople.org/groups/cockpit/logs/tests-data-10276-20181011-123232/

**Comment by martinpitt** [comment]
That [exception](http://fedorapeople.org/groups/cockpit/logs/tests-data-10276-20181011-123232/) seems to be PR #10279. I just landed that, so next run should look better?

**Comment by cockpituous** [comment]
learn-tests in progress on centosci-tasks-db661.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181011-132727/

**Comment by cockpituous** [comment]
https://cockpit-learn.cockpit.svc.cluster.localNone/train/tests-train-1.jsonl.gz: couldn't push training data to learn service: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181011-132727/

**Comment by stefwalter** [comment]
Failure because  #10278 is not merged yet. Retrying.

**Comment by cockpituous** [comment]
learn-tests in progress on bos-tasks-b5324.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181011-133838/

**Comment by cockpituous** [comment]
https://cockpit-learn.verify.svc.cluster.localNone/train/tests-train-1.jsonl.gz: couldn't push training data to learn service: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181011-133838/

**Comment by cockpituous** [comment]
learn-tests in progress on centosci-tasks-p0lrh.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181011-133838/

**Comment by cockpituous** [comment]
https://cockpit-learn.cockpit.svc.cluster.localNone/train/tests-train-1.jsonl.gz: couldn't push training data to learn service: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181011-133838/

**Comment by cockpituous** [comment]
learn-tests in progress on bos-tasks-2xs1k.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181011-134545/

**Comment by cockpituous** [comment]
https://cockpit-learn.verify.svc.cluster.localNone/train/tests-train-1.jsonl.gz: couldn't push training data to learn service: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181011-134545/

**Comment by stefwalter** [comment]
There seems to be a bug that does not like the  colon in the urls. Will look into it.

**Comment by stefwalter** [comment]
learn-tests in progress on falcon.
Log: http://stefw.fedorapeople.org/logs/learn-tests-10276-20181011-161616/log

**Comment by stefwalter** [comment]
FAIL: http://stefw.fedorapeople.org/logs/learn-tests-10276-20181011-161616/log

**Comment by stefwalter** [comment]
learn-tests in progress on falcon.
Log: http://stefw.fedorapeople.org/logs/learn-tests-10276-20181011-161818/log

**Comment by stefwalter** [comment]
FAIL: http://stefw.fedorapeople.org/logs/learn-tests-10276-20181011-161818/log

**Comment by cockpituous** [comment]
learn-tests in progress on centosci-tasks-6qmvj.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181011-150101/

**Comment by cockpituous** [comment]
https://cockpit-learn.cockpit.svc.cluster.localNone/train/tests-train-1.jsonl.gz: couldn't push training data to learn service: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181011-150101/

**Comment by cockpituous** [comment]
learn-tests in progress on bos-tasks-69xmw.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181011-150808/

**Comment by cockpituous** [comment]
https://cockpit-learn.verify.svc.cluster.localNone/train/tests-train-1.jsonl.gz: couldn't push training data to learn service: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181011-150808/

File: bots/learn-tests [hunk 1]
```diff
@@ -26,89 +26,110 @@ TRAINING_DATA = "tests-train-1.jsonl.gz"
...
+        filename = os.path.join(DATA, filename)
+    train(filename, url, verbose)
+
+# Does 'tail -F' on an HTTP URL
+def tail(url, until, verbose=False):
+    stop = False
+    at = 0
+
+    while True:
+        print("xxx", url, verbose)
```
Comment (Gundersanne) [inline bots/learn-tests]: Lingering debug print?

Comment (stefwalter) [reply bots/learn-tests]: Removed.

**Comment by stefwalter** [comment]
> I'm guessing the actual learning service isn't running yet?

It is, i'll need to figure out why it's failing.

**Comment by cockpituous** [comment]
learn-tests in progress on centosci-tasks-db661.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181011-174747/

**Comment by cockpituous** [comment]
https://cockpit-learn.cockpit.svc.cluster.local:8080/train/tests-train-1.jsonl.gz: couldn't push training data to learn service: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181011-174747/

**Comment by cockpituous** [comment]
learn-tests in progress on centosci-tasks-db661.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181011-175252/

**Comment by cockpituous** [comment]
learn-tests in progress on bos-tasks-nvdz7.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181011-185050/

**Comment by stefwalter** [comment]
Yay working now.

**Comment by cockpituous** [comment]
learn-tests in progress on centosci-tasks-p0lrh.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181012-083636/

**Comment by cockpituous** [comment]
learn-tests in progress on bos-tasks-69xmw.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181012-093939/

**Comment by cockpituous** [comment]
learn-tests in progress on centosci-tasks-p0lrh.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181012-105757/

**Comment by cockpituous** [comment]
learn-tests in progress on bos-tasks-8frq3.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181012-115252/

**Comment by cockpituous** [comment]
Task aborted.

**Comment by cockpituous** [comment]
learn-tests in progress on bos-tasks-m7v76.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181012-125252/

**Comment by stefwalter** [comment]
Triggering again on purpose to test another iteration on the containers.

**Comment by cockpituous** [comment]
learn-tests in progress on centosci-tasks-6qmvj.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181012-170606/

**Comment by cockpituous** [comment]
learn-tests in progress on bos-tasks-m7v76.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181012-171212/

**Comment by cockpituous** [comment]
learn-tests in progress on bos-tasks-nvdz7.
Log: http://fedorapeople.org/groups/cockpit/logs/learn-tests-10276-20181012-182020/

**Comment by stefwalter** [comment]
Fixed issue in tests-policy, testing again to see if retries work in the test results.

File: bots/learn-tests (hunk 1 above)
Comment (Gundersanne) [reply bots/learn-tests]: Got added again.

**Pull request merged**